### DISCUSSION TITLE
cppflow::tensor from pointer to void

### DISCUSSION THREAD

**Discussion by 8-byte** [serizba/cppflow#186] [Q&A] [answered]
Hello everyone,
I would like to construct a cppflow::tensor directly from an existing buffer already in memory without having to create an additional intermediary std::vector of the values. There is already a constructor I can use that takes in the data type, a void pointer, the length of the buffer, and the shape, but it has been made private in the current implementation. Is there a reason why we wouldn't want this constructor to be public?

**Comment by serizba** [comment, marked as answer]
Hi @8-byte 

You can use that constructor directly, there shouldn't be any problem. 

I made it private because it's not as encapsulated as the others, and it provides a lower level of abstraction. I thought that if you need to use this function, you understand the code well enough to modify it and make it public (like your case). 

On the other hand, for novel users, it may be confusing to choose what constructor to use and how, and that's why I decided to only expose the abstract constructors.

**Comment by 8-byte** [comment]
Hi @serizba,
Thanks for the prompt reply. That makes sense. I think exposing that constructor in a future release could make the interface more complete since there are many use cases where the data already resides in memory and creating an intermediary std::vector may introduce unwanted overhead. Thanks again!
